src/clang-c-frontend/clang_c_convert.h

```diff
@@ -206,6 +206,8 @@ class clang_c_convertert
   const clang::Decl *get_top_FunctionDecl_from_Stmt(const clang::Stmt &stmt);
 
   void gen_typecast_to_union(exprt &dest, const typet &type);
+
+  static constexpr const char *mode = "C";
```

Comment:
Can we make it a protected class member? That way we can initialize it with "C++" in the cpp frontend and get rid of the `get_default_symbol` overload.

Reply:
@kunjsong01 there is no need to pass the string as an argument to the constructor, simply initialize it in the initializer list.

Reply:
@mikhailramalho How do we initialize clang_c_converter's `mode`  in clang_cpp_converter?  
or do we want a `mode` member in clang_c_converter and clang_cpp_converter respectively?

Reply:
I meant something like:
```
diff --git a/src/clang-c-frontend/clang_c_convert.cpp b/src/clang-c-frontend/clang_c_convert.cpp
index ed7ea6e6c..72873227e 100644
--- a/src/clang-c-frontend/clang_c_convert.cpp
+++ b/src/clang-c-frontend/clang_c_convert.cpp
@@ -28,11 +28,13 @@
 
 clang_c_convertert::clang_c_convertert(
   contextt &_context,
-  std::vector<std::unique_ptr<clang::ASTUnit>> &_ASTs)
+  std::vector<std::unique_ptr<clang::ASTUnit>> &_ASTs,
+  const char *_mode)
   : ASTContext(nullptr),
     context(_context),
     ns(context),
     ASTs(_ASTs),
+    mode(_mode),
     anon_symbol("clang_c_convertert::"),
     current_scope_var_num(1),
     current_block(nullptr),
@@ -41,6 +43,13 @@ clang_c_convertert::clang_c_convertert(
 {
 }
 
+clang_c_convertert::clang_c_convertert(
+  contextt &_context,
+  std::vector<std::unique_ptr<clang::ASTUnit>> &_ASTs)
+  : clang_c_convertert(_context, _ASTs, "C")
+{
+}
+
 bool clang_c_convertert::convert()
 {
   if(convert_top_level_decl())
diff --git a/src/clang-c-frontend/clang_c_convert.h b/src/clang-c-frontend/clang_c_convert.h
index c0b7e20de..6bc94a16f 100644
--- a/src/clang-c-frontend/clang_c_convert.h
+++ b/src/clang-c-frontend/clang_c_convert.h
@@ -65,10 +65,16 @@ public:
   static std::string get_decl_name(const clang::NamedDecl &nd);
 
 protected:
+  clang_c_convertert(
+    contextt &_context,
+    std::vector<std::unique_ptr<clang::ASTUnit>> &_ASTs,
+    const char *_mode);
+
   clang::ASTContext *ASTContext;
   contextt &context;
   namespacet ns;
   std::vector<std::unique_ptr<clang::ASTUnit>> &ASTs;
+  const char *mode;
 
   symbol_generator anon_symbol;
 
diff --git a/src/clang-cpp-frontend/clang_cpp_convert.cpp b/src/clang-cpp-frontend/clang_cpp_convert.cpp
index 079f62881..b3d23ff82 100644
--- a/src/clang-cpp-frontend/clang_cpp_convert.cpp
+++ b/src/clang-cpp-frontend/clang_cpp_convert.cpp
@@ -26,7 +26,7 @@
 clang_cpp_convertert::clang_cpp_convertert(
   contextt &_context,
   std::vector<std::unique_ptr<clang::ASTUnit>> &_ASTs)
-  : clang_c_convertert(_context, _ASTs)
+  : clang_c_convertert(_context, _ASTs, "C++")
 {
 }
```

I'm not sure I like it better tho. @fbrausse what do you think?

Reply:
Btw, I already think the PR is good to merge, this is just cosmetics.